terraform/install.sh

```diff
@@ -278,32 +290,72 @@ checkAuthentication() {
         log "Authentication failed"
         log "Please allow gcloud and Cloud Shell to access your GCP account"
     fi
-    while [[ -z $AUTH_ACCT  && "${TRIES}" -lt 10  ]]; do
+    while [[ -z $AUTH_ACCT  && "${TRIES}" -lt 30  ]]; do
         AUTH_ACCT=$(gcloud auth list --format="value(account)")
-        sleep 3;
+        sleep 1;
         TRIES=$((TRIES + 1))
     done
     if [[ -z $AUTH_ACCT ]]; then
         exit 1
     fi
 }
 
-log "Checking Prerequisites..."
+parseArguments() {
+  while (( "$#" )); do
+    case "$1" in
+    -p|--project|--project-id)
+      if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+        project_id=$2
```

Comment:
we should also check that projectid is not empty

Reply:
This is already captured in `[ -n "$2" ]`. If the value is empty, it will print "Error: Argument for --project-id is missing"